File: codigo/python3/02_tipos_de_datos_escalares_en_python/02_tiempo_calculo.py
import time
import fractions
from math import log

print('ln(2) [exacto] = ', log(2))

# cálculo con fractions (truncado a 200.000 términos)
suma = 0
tic = time.time()
for i in range(0, 100_000):
   suma += fractions.Fraction(1, 2*i + 1) - fractions.Fraction(1, 2*i + 2)
toc = time.time()
print('ln(2) = ', float(suma))
print (f'El cálculo se realizó en {toc-tic} segundos')

# cálculo con floats (truncado a 200.000 términos)
suma = 0
tic = time.time()
for i in range(0, 100_000):
   suma += 1/(2*i + 1) - 1/(2*i + 2);
toc = time.time()
print('ln(2) = ', suma)
print (f'El cálculo se realizó en {toc-tic} segundos')

File: codigo/python3/02_tipos_de_datos_escalares_en_python/02_tiempo_calculo.c
#include <stdio.h>
#include <sys/time.h>
#include <unistd.h>

double intervalo_tiempo_ms(struct timeval toc, struct timeval tic);

int main() {
   struct timeval tic, toc;
   double x, dif, suma = 0;

   gettimeofday(&tic, NULL);   
   for (int i=0; i < 10000000; i++)
      suma += 1.0/(2*i + 1) - 1.0/(2*i + 2);
   gettimeofday(&toc, NULL);

   dif = intervalo_tiempo_ms(toc, tic);

   printf("ln(2) = %g\n", suma);
   printf("El cálculo se realizó en %g segundos.\n", dif);

   return 0;
}

double intervalo_tiempo_ms(struct timeval toc, struct timeval tic)
{
   return (toc.tv_sec - tic.tv_sec) + (toc.tv_usec - tic.tv_usec)/1000000.0;
}
